apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4CoverageResourceProvider.java

```diff
@@ -64,38 +65,35 @@ public final class R4CoverageResourceProvider implements IResourceProvider {
   /** The entity manager. */
   private EntityManager entityManager;
   /** The Metric registry. */
-  private MetricRegistry metricRegistry;
+  private final MetricRegistry metricRegistry;
   /** The Loaded filter manager. */
-  private LoadedFilterManager loadedFilterManager;
+  private final LoadedFilterManager loadedFilterManager;
 
   /**
-   * Sets the {@link #entityManager}.
+   * Instantiates a new {@link R4CoverageResourceProvider}.
    *
-   * @param entityManager a JPA {@link EntityManager} connected to the application's database
-   */
-  @PersistenceContext
-  public void setEntityManager(EntityManager entityManager) {
-    this.entityManager = entityManager;
-  }
-
-  /**
-   * Sets the {@link #metricRegistry}.
+   * <p>Spring will wire this class during the initial component scan, so this constructor should
+   * only be explicitly called by tests.
    *
-   * @param metricRegistry the {@link MetricRegistry} to use
+   * @param metricRegistry the metric registry
+   * @param loadedFilterManager the loaded filter manager
    */
-  @Inject
-  public void setMetricRegistry(MetricRegistry metricRegistry) {
+  public R4CoverageResourceProvider(
+      MetricRegistry metricRegistry, LoadedFilterManager loadedFilterManager) {
+    requireNonNull(metricRegistry);
+    requireNonNull(loadedFilterManager);
     this.metricRegistry = metricRegistry;
+    this.loadedFilterManager = loadedFilterManager;
   }
 
   /**
-   * Sets the {@link #loadedFilterManager}.
+   * Sets the {@link #entityManager}.
    *
-   * @param loadedFilterManager the {@link LoadedFilterManager} to use
+   * @param entityManager a JPA {@link EntityManager} connected to the application's database
    */
-  @Inject
-  public void setLoadedFilterManager(LoadedFilterManager loadedFilterManager) {
-    this.loadedFilterManager = loadedFilterManager;
+  @PersistenceContext
+  public void setEntityManager(EntityManager entityManager) {
+    this.entityManager = entityManager;
```

Comment:
Just curious, why is this not done through constructor injection too?

Reply:
Throwing the persistence context in the constructor params wasn't working as-is, so I left it. It might be possible but I didn't mess with it a ton, since it seemed like it needed some more complicated setup.